src/System.Net.Sockets/src/System/Net/Sockets/Socket.Unix.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Globalization;
+using System.IO;
+using System.Net;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Diagnostics;
+using System.Diagnostics.Contracts;
+using System.Threading.Tasks;
+
+namespace System.Net.Sockets
+{
+    public partial class Socket
+    {
+        internal static int FillFdSetFromSocketList(ref Interop.libc.fd_set fdset, IList socketList)
```

Comment:
I realize this would affect upstream code as well, but could the argument be changed to be an ```IList<Socket>``` instead?